From 3fe75d92883d43a2a0663745faaacb80eaecfa5d Mon Sep 17 00:00:00 2001
From: Chih-Hsuan Yen <yan12125@gmail.com>
Date: Fri, 17 Jun 2022 19:56:02 +0800
Subject: [PATCH]  Allow tests to pass without XBR dependencies (#1580)

* Allow tests to pass without XBR dependencies - fixes #1579
* Address review comments
* Fix test skipping notes
---
 autobahn/xbr/test/test_xbr_frealm.py              | 11 ++++++-----
 autobahn/xbr/test/test_xbr_schema_demo.py         |  7 ++++++-
 autobahn/xbr/test/test_xbr_schema_wamp.py         | 12 +++++++++---
 autobahn/xbr/test/test_xbr_schema_wamp_control.py |  7 ++++++-
 autobahn/xbr/test/test_xbr_secmod.py              | 10 +++++-----
 5 files changed, 32 insertions(+), 15 deletions(-)

diff --git a/autobahn/xbr/test/test_xbr_frealm.py b/autobahn/xbr/test/test_xbr_frealm.py
index 5e511a9d..d85a78ef 100644
--- a/autobahn/xbr/test/test_xbr_frealm.py
+++ b/autobahn/xbr/test/test_xbr_frealm.py
@@ -7,11 +7,12 @@ from twisted.trial.unittest import TestCase
 from twisted.internet.defer import inlineCallbacks
 
 from autobahn.xbr import HAS_XBR
-if HAS_XBR:
-    from autobahn.xbr._frealm import Seeder, FederatedRealm
+from autobahn.wamp.cryptosign import HAS_CRYPTOSIGN
 
-from autobahn.xbr._secmod import SecurityModuleMemory, EthereumKey
-from autobahn.wamp.cryptosign import CryptosignKey
+if HAS_XBR and HAS_CRYPTOSIGN:
+    from autobahn.xbr._frealm import Seeder, FederatedRealm
+    from autobahn.xbr._secmod import SecurityModuleMemory, EthereumKey
+    from autobahn.wamp.cryptosign import CryptosignKey
 
 # https://web3py.readthedocs.io/en/stable/providers.html#infura-mainnet
 HAS_INFURA = 'WEB3_INFURA_PROJECT_ID' in os.environ and len(os.environ['WEB3_INFURA_PROJECT_ID']) > 0
@@ -22,7 +23,7 @@ IS_CPY_310 = sys.version_info.minor == 10
 
 @skipIf(not os.environ.get('USE_TWISTED', False), 'only for Twisted')
 @skipIf(not HAS_INFURA, 'env var WEB3_INFURA_PROJECT_ID not defined')
-@skipIf(not HAS_XBR, 'package autobahn[xbr] not installed')
+@skipIf(not (HAS_XBR and HAS_CRYPTOSIGN), 'package autobahn[encryption,xbr] not installed')
 class TestFederatedRealm(TestCase):
 
     gw_config = {
--- a/autobahn/xbr/test/test_xbr_secmod.py
+++ b/autobahn/xbr/test/test_xbr_secmod.py
@@ -34,15 +34,15 @@ from unittest import skipIf
 from twisted.internet.defer import inlineCallbacks
 from twisted.trial.unittest import TestCase
 
-from py_eth_sig_utils.eip712 import encode_typed_data
-from py_eth_sig_utils.utils import ecsign, ecrecover_to_pub, checksum_encode, sha3
-from py_eth_sig_utils.signing import v_r_s_to_signature, signature_to_v_r_s
-from py_eth_sig_utils.signing import sign_typed_data, recover_typed_data
-
 from autobahn.wamp.cryptosign import HAS_CRYPTOSIGN
 from autobahn.xbr import HAS_XBR
 
 if HAS_XBR and HAS_CRYPTOSIGN:
+    from py_eth_sig_utils.eip712 import encode_typed_data
+    from py_eth_sig_utils.utils import ecsign, ecrecover_to_pub, checksum_encode, sha3
+    from py_eth_sig_utils.signing import v_r_s_to_signature, signature_to_v_r_s
+    from py_eth_sig_utils.signing import sign_typed_data, recover_typed_data
+
     from autobahn.xbr import make_w3, EthereumKey, mnemonic_to_private_key
     from autobahn.xbr._eip712_member_register import _create_eip712_member_register
     from autobahn.xbr._eip712_market_create import _create_eip712_market_create
-- 
2.36.1

